js/authentication.md

```diff
@@ -561,6 +561,115 @@ export default class App extends React.Component {
 }
 ```
 
+#### Federated with Auth0
+
+You can use `Auth0` as one of the provider of your Cognito Federated Identity Pool so that you can allow your Auth0 users to have access to your AWS resources. 
+
+Step 1. Learn [how to integrate Auth0 with Cognito Federated Identity Pool](https://auth0.com/docs/integrations/integrating-auth0-amazon-cognito-mobile-apps)
+
+Step 2. Login with `Auth0` and then use the id token to get the AWS credentials from `Cognito Federated Idneitty Pool` by `Auth.federatedSignIn`, for example:
+```js
+const { idToken, domain, expiresIn, user } = getFromAuth0();
+
+Auth.federatedSignIn({
+    domain, // The Auth0 Domain,
+    {
+        token: idToken // The id token from Auth0
+        expires_at: expiresIn * 1000 + new Date().getTime() // the expiration timestamp
+    },
+    user // the user object, e.x. { name: username, email: email }
+}).then(cred => {
+    console.log(cred);
+});
+```
+
+Step 3. Now you can get the current user and current Credentials:
+```js
+Auth.currentAuthenticatedUser().then(user => console.log(user));
+Auth.currentCredentials().then(creds => console.log(creds));
+
+// Unfortunately Auth.currentSession() doesn't support federated identities for now. Please store the auth0 session info manually.
+```
+
+Step 4. If you want the credentials not to get expired, you can pass a refresh handler to the Auth module to refresh the id token from `Auth0`:
```

Comment:
Step 4. You can pass a refresh handler to the Auth module to refresh the id token from `Auth0`

Reply:
Done.